server/service/src/preference/upsert_helpers.rs

```diff
@@ -0,0 +1,54 @@
+use repository::{PreferenceRow, PreferenceRowRepository, StorageConnection};
+
+use crate::sync::CentralServerConfig;
+
+use super::UpsertPreferenceError;
+
+pub fn upsert_global(
```

Comment:
Like query, these are the default upsert helpers. If need be, can implement your own upsert, e.g. if you need more validation logic